### PR TITLE
Audio file list position fix

### PR DESCRIPTION
**Pull request by kookster**
- [x] Remove `acts_as_list` from `AudioFile`, leave it on other models
- [x] Add some defaulting of position on `AudioFile` when position is null
- [x] Add validation on `AudioVersion` to make sure audio file positions have no gaps

### CODE REVIEW

File: app/models/audio_version.rb [hunk 1]
```diff
@@ -64,6 +64,13 @@ def set_status
       return
     end
 
+    pos = 0
+    if audio_files.detect { |af| af.position != (pos += 1) }
+      self.status = INVALID
+      self.status_message = "Audio file missing for position #{pos}"
```
Comment (cavis) [inline app/models/audio_version.rb]: It doesn't actually matter.  But I might prefer this error be _after_ the audio-version-template errors in priority.  Meaning you'd get the current "Audio version has 2 files but must have 3 files" error before this one.

File: test/factories/audio_version_factory.rb [hunk 2]
```diff
@@ -15,7 +15,8 @@
     after(:create) do |audio_version, evaluator|
       c = evaluator.audio_files_count
       c = (rand(5) + 1) if (evaluator.audio_files_count == 0)
-      FactoryGirl.create_list(:audio_file, c, audio_version: audio_version)
+      c = 0 if evaluator.audio_files_count < 0
```
Comment (cavis) [inline test/factories/audio_version_factory.rb]: So now < 0 means 0, and 0 means > 0?

File: app/models/audio_version.rb (hunk 1 above)
Comment (kookster) [reply app/models/audio_version.rb]: thanks, I think you're right - I'll move it.

File: test/factories/audio_version_factory.rb (hunk 2 above)
Comment (kookster) [reply test/factories/audio_version_factory.rb]: just checked, no place in the code is using the current 0 value, so getting rid of the random, and simplifying this.

File: app/models/audio_version.rb [hunk 3]
```diff
@@ -72,5 +72,12 @@ def set_status
       self.status = INVALID
       self.status_message = errors
```
Comment (cavis) [inline app/models/audio_version.rb]: You probably need a `return` here now, right?

Comment (kookster) [reply app/models/audio_version.rb]: argh, yes!

**Comment by kookster** [comment]
@cavis that should do it

**Comment by cavis** [comment]
:+greatest:

**Pull request merged**